### PR TITLE
Convert to input - exp-ed, search

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: -v -k "expression or advanced_search" --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0acab6d5dc0f50ac541b0ad45f3e012f8e208b09:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by seandst** [comment]
Quite a few failures to locate inputs in the tester logs, which might be a problem with Input itself. Can you look into some of the examples and fix Input up?

**Comment by mfalesni** [comment]
@seandst Sure, going into it.

**Comment by dajoRH** [comment]
Lint report for commit 371c448a0e235d1701e4f1fcadd0a88752a7da63:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c937a56848f112a0502b74470090decc734527a5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@seandst I changed the comment ... and added quoteattr to the name attribute to make it absolutely bulletproof.

**Comment by seandst** [comment]
> @seandst I changed the comment ... and added quoteattr to the name attribute to make it absolutely bulletproof.

CFME will find a way... :(

What's up with the "NoSuchElementException: Message: Element <cfme.web_ui.Input name='search[text]'> not found on page." in PRT results?

**Comment by mfalesni** [comment]
@seandst I'll check.

**Comment by dajoRH** [comment]
Lint report for commit 28398e22b04a528e0441a36a448f57127147b399:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e56b1b221d435cf237757cc8a776f6f3ec5bad7a:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/web_ui/__init__.py [hunk 1]
```diff
@@ -1239,13 +1241,22 @@ def _fill_form_dict(form, values, **kwargs):
...
+    pretty_attrs = ['names']
+
+    def __init__(self, *names):
+        self.names = names
 
     def locate(self):
-        return '//input[@name="{}"]'.format(self.name)
+        return '(//input | //textarea)[{}]'.format(
+            " or ".join("@name={}".format(quoteattr(name)) for name in self.names)
+        )
```
Comment (mfalesni) [inline cfme/web_ui/__init__.py]: @seandst This was required due to search boxes in host and vms having slightly different name (search_text vs search[text])

**Comment by dajoRH** [comment]
Lint report for commit 3a03601d9a9be23f68947f940a1a137759dcbb0d:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c16e33f3fb2eb01b40543191407d53c3d8b73fe2:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f2fb7341b498b3b20fb49b115dfdd22aa4f90abc:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 760f0c4c51b249dfd06a2d882badbd5e61f6566c:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./conftest.py:11:1: F401 'AuthSetting' imported but unused
```
*CFME QE Bot*

**Comment by mfalesni** [comment]
Seems this will require some fiddling around and apparently there are some serious issues with advanced search in new code.

**Comment by mfalesni** [comment]
Although I tried to poke the advanced search, I'd personally say to let it go for now and see if it stabilizes and then tweak the fixes.

**Comment by dajoRH** [comment]
Lint report for commit 32930e766a95f88ca64b1ba9a5b1c03fcf002e7e:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**